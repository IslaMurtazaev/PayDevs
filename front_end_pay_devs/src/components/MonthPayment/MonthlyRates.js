import React, { Component } from "react";
import { connect } from "react-redux";

import monthPaymentActions from "../../actions/monthPayment";
import MonthlyRate from "./MonthlyRate";
import CreateMonthPaymentForm from "./CreateMonthPaymentForm";

class MonthlyRates extends Component {
  componentDidMount() {
    if (!this.props.monthPayments.length)
      this.props.getAllMonthPayments(this.props.projectId);
  }

  render() {
    const { monthPayments, projectId, removeMonthPayment } = this.props;

    return (
      <div>
        {monthPayments.length > 0 && (
          <h3 className="rateHeader">Select one of your current rates:</h3>
        )}
        <div>
          {monthPayments.map(monthPayment => (
            <MonthlyRate
              key={monthPayment.id}
              monthPayment={monthPayment}
              projectId={projectId}
              onRemove={monthPaymentId =>
                removeMonthPayment(monthPaymentId)
              }
            />
          ))}
        </div>

        <hr />

        <CreateMonthPaymentForm projectId={projectId} />
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    monthPayments: state.monthPayments
  };
};

const mapDispatchersToProps = dispatch => ({
  getAllMonthPayments: projectId =>
    dispatch(monthPaymentActions.getAll(projectId)),
  removeMonthPayment: monthPaymentId =>
    dispatch(monthPaymentActions.remove(monthPaymentId))
});

export default connect(
  mapStateToProps,
  mapDispatchersToProps
)(MonthlyRates);
