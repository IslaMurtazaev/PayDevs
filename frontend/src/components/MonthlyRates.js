import React, { Component } from "react";

import MonthlyRate from "./MonthlyRate";
import CreateMonthPaymentForm from "./CreateMonthPaymentForm";

class MonthlyRates extends Component {
  componentWillMount() {
    let { monthPayments, projectId, getAllMonthPayments } = this.props;
    if (!monthPayments.length || monthPayments[0].projectId !== projectId)
      getAllMonthPayments(projectId);
  }

  render() {
    const { monthPayments, projectId, removeMonthPayment } = this.props;

    return (
      <div className="monthlyRates">
        {monthPayments.length > 0 && (
          <h3 className="rateHeader">Select one of your current rates:</h3>
        )}
        <div>
          {monthPayments.map(monthPayment => (
            <MonthlyRate
              key={monthPayment.id}
              monthPayment={monthPayment}
              projectId={projectId}
              onRemove={monthPaymentId => removeMonthPayment(monthPaymentId)}
            />
          ))}
        </div>

        <CreateMonthPaymentForm projectId={projectId} />
      </div>
    );
  }
}

export default MonthlyRates;
