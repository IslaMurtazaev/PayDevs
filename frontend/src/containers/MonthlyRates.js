import { connect } from "react-redux";

import monthPaymentActions from "../actions/monthPayment";
import MonthlyRates from "../components/MonthlyRates"

const mapStateToProps = state => {
  return {
    monthPayments: state.monthPayments
  };
};

const mapDispatchersToProps = dispatch => ({
  getAllMonthPayments: projectId =>
    dispatch(monthPaymentActions.getAll(projectId)),
  removeMonthPayment: monthPaymentId =>
    dispatch(monthPaymentActions.remove(monthPaymentId))
});

export default connect(
  mapStateToProps,
  mapDispatchersToProps
)(MonthlyRates);
