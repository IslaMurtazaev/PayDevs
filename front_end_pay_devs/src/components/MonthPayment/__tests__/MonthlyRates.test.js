import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import configureStore from "redux-mock-store";

import MonthlyRates from "../MonthlyRates";

describe("<MonthlyRates />", () => {
  const initialState = {
    monthPayments: [
      { id: 1, rate: 150 },
      { id: 2, rate: 200 },
      { id: 3, rate: 220 }
    ]
  };

  const mockStore = configureStore();
  const store = mockStore(initialState);

  let container;
  beforeEach(() => {
    container = shallow(<MonthlyRates store={store} />).dive();
  });

  it("renders rateHeader if there are props.monthPayments", () => {
    expect(container.find(".rateHeader").text()).toBe(
      "Select one of your current rates:"
    );
  });

  it("matches previous snapshot", () => {
    expect(toJson(container)).toMatchSnapshot();
  });
});
