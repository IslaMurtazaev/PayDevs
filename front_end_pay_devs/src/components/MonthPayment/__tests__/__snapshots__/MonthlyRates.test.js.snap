// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MonthlyRates /> matches previous snapshot 1`] = `
<div
  className="monthlyRates"
>
  <h3
    className="rateHeader"
  >
    Select one of your current rates:
  </h3>
  <div>
    <MonthlyRate
      key="1"
      monthPayment={
        Object {
          "id": 1,
          "rate": 150,
        }
      }
      onRemove={[Function]}
    />
    <MonthlyRate
      key="2"
      monthPayment={
        Object {
          "id": 2,
          "rate": 200,
        }
      }
      onRemove={[Function]}
    />
    <MonthlyRate
      key="3"
      monthPayment={
        Object {
          "id": 3,
          "rate": 220,
        }
      }
      onRemove={[Function]}
    />
  </div>
  <Connect(CreateMonthPaymentForm) />
</div>
`;
